rio_tiler/io/stac.py

```diff
@@ -306,6 +307,14 @@ def _get_asset_info(self, asset: str) -> AssetInfo:
                 for b in bands
                 if {"minimum", "maximum"}.issubset(b.get("statistics", {}))
             ]
+            # check that stats data are all double and make warning if not
+            if stats and not all(
+                isinstance(v, (int, float)) for stat in stats for v in stat
+            ):
+                warnings.warn(
+                    "Some statistics data are not double, this may cause issues in data processing."
+                )
+
```

Comment:
I'm not sure it makes 💯 sense just to raise a warning because this will likely create an Error later. 

I think we could raise a warning AND not add the `invalid` statistics.

Reply:
sure. that is even better